src/couch/src/couch_file.erl

```diff
@@ -596,7 +604,28 @@ handle_call({write_header, Bin}, _From, #file{fd = Fd, eof = Pos} = File) ->
             {reply, Error, reset_eof(File)}
     end;
 handle_call(find_header, _From, #file{fd = Fd, eof = Pos} = File) ->
-    {reply, find_header(Fd, Pos div ?SIZE_BLOCK), File}.
+    {reply, find_header(Fd, Pos div ?SIZE_BLOCK), File};
+handle_call({commit_header, Bin}, From, #file{fd = Fd} = File) ->
+    case fsync(Fd) of
```

Comment:
a `maybe` expression would be clearer and is available in all versions of OTP that couchdb supports.

Reply:
rewritten here c4d538a232130b11366fe4ede459100d11ba5d2b